Source/Scene/PointCloud3DTileContent.js

```diff
@@ -823,58 +713,310 @@ define([
+        for (var i = 0; i < numberOfAttributes; ++i) {
+            var attribute = vertexArray.getAttribute(i);
+            if (attribute.index === index) {
+                return attribute;
+            }
+        }
+    }
+
+    function createShaders(content, frameState, style) {
+        var i;
+        var name;
+        var attribute;
+
+        var context = frameState.context;
+        var batchTable = content.batchTable;
+        var hasBatchTable = defined(batchTable);
+        var hasStyle = defined(style);
+        var isQuantized = content._isQuantized;
+        var isOctEncoded16P = content._isOctEncoded16P;
+        var isRGB565 = content._isRGB565;
+        var isTranslucent = content._isTranslucent;
+        var hasColors = content._hasColors;
+        var hasNormals = content._hasNormals;
+        var hasBatchIds = content._hasBatchIds;
+        var backFaceCulling = content._backFaceCulling;
+        var vertexArray = content._drawCommand.vertexArray;
+
+        var colorStyleFunction;
+        var showStyleFunction;
+        var styleTranslucent = isTranslucent;
+
+        if (hasBatchTable) {
+            // Styling is handled in the batch table
+            hasStyle = false;
+        }
+
+        if (hasStyle) {
+            var shaderState = {
+                translucent : false
+            };
+            colorStyleFunction = style.getColorShaderFunction('getColorFromStyle', 'czm_tiles3d_style_', shaderState);
+            showStyleFunction = style.getShowShaderFunction('getShowFromStyle', 'czm_tiles3d_style_', shaderState);
+            styleTranslucent = shaderState.translucent;
+        }
+
+        content._styleTranslucent = styleTranslucent;
+
+        var hasColorStyle = defined(colorStyleFunction);
+        var hasShowStyle = defined(showStyleFunction);
+
+        // Get the properties in use by the style
+        var styleableProperties = [];
+        if (hasColorStyle) {
+            getStyleableProperties(colorStyleFunction, styleableProperties);
+        }
+        if (hasShowStyle) {
+            getStyleableProperties(showStyleFunction, styleableProperties);
+        }
+
+        // Disable vertex attributes that aren't used in the style, enable attributes that are
+        var styleableShaderAttributes = content._styleableShaderAttributes;
+        if (defined(styleableShaderAttributes)) {
+            for (name in styleableShaderAttributes) {
+                if (styleableShaderAttributes.hasOwnProperty(name)) {
+                    attribute = styleableShaderAttributes[name];
+                    var enabled = (styleableProperties.indexOf(name) >= 0);
+                    var vertexAttribute = getVertexAttribute(vertexArray, attribute.location);
+                    vertexAttribute.enabled = enabled;
+                }
+            }
+        }
+
+        if (hasColorStyle) {
+            // Disable the color vertex attribute when a color style is used
```

Comment:
Are you sure we should disable this?  Couldn't this be used in a styling expression?